cranelift-filetests/src/function_runner.rs

```diff
@@ -76,17 +298,161 @@ impl FunctionRunner {
+        if !(normal_rets[0].value_type == types::I64 && normal_rets[1].value_type == types::I64) {
+            return Err(String::from(
+                "Function return type with two parameters can only return i64, i64.",
+            ));
+        }
+
+        return Ok(types::I128);
+    }
+    Ok(normal_rets[0].value_type)
+}
+
+impl FunctionRunnerBuilder {
+    /// Create a new function runner builder with the specified function.
+    pub fn new(function: Function) -> Self {
+        Self {
+            action: FunctionRunnerAction::Test,
+            function: function,
+            isa: None,
+        }
+    }
+
+    /// Use the specified action.
+    pub fn with_action(mut self, action: FunctionRunnerAction) -> Self {
+        self.action = action;
+        self
+    }
+
+    /// Parse and use the specified action.
+    pub fn with_action_text(mut self, action: &str) -> Self {
+        self.action = Self::parse_action(action, return_type(&self.function).unwrap());
+        self
+    }
+
+    /// Use the specified ISA (must be the host machine's ISA).
+    pub fn with_isa(mut self, isa: Box<dyn TargetIsa>) -> Self {
+        self.isa = Some(isa);
+        self
+    }
+
+    /// Use the host ISA with the specified flags.
+    pub fn with_host_isa(self, flags: settings::Flags) -> Self {
+        let builder = host_isa_builder().expect("Unable to build a TargetIsa for the current host");
+        self.with_isa(builder.finish(flags))
+    }
+
+    /// Use the default host ISA.
+    pub fn with_default_host_isa(self) -> Self {
+        let flags = settings::Flags::new(settings::builder());
+        self.with_host_isa(flags)
+    }
+
+    /// Build a function runner from an action, a function and the ISA to run on.
+    pub fn finish(self) -> FunctionRunner {
+        FunctionRunner::new(self.function, self.isa.unwrap(), self.action)
+    }
+
+    /// Parse a function runner action from a comment string such as "; run print".
+    fn parse_action(comment: &str, ty: types::Type) -> FunctionRunnerAction {
+        let comment = comment.trim();
+        let comment = comment.trim_start_matches(";");
+        let comment = comment.trim_start();
+        let comment = comment.trim_start_matches("run");
+        let comment = comment.trim_start_matches(":");
+        let comment = comment.trim_start();
+
+        if comment.len() == 0 {
+            // Don't print test return value.
+            return FunctionRunnerAction::Test;
+        }
+
+        if comment == "print" {
+            // Print test return value, if return type is not b1, always fail the test.
+            return FunctionRunnerAction::Print;
+        }
+
+        // ";   run:  %fn() > 42 " is now "%fn() > 42".
+        let mut parts: Vec<&str> = comment.split_ascii_whitespace().collect();
+
+        if parts.len() < 3 {
+            panic!("Invalid syntax for test assertion, expected \"%fn() <operator> <value>\", found \"{}\"", comment);
```

Comment:
We shouldn't panic on parse errors, we should return a fallible `ParseResult` like other parsing functions do.

Additionally, it seems a little strange that we have this parsing code in the `cranelift-filetests` crate, when AFAICT all the other parsing code is in the `cranelift-reader` crate. Unless there is a good reason not to, we should match existing conventions and keep the parsing code inside `cranelift-reader`.